kedro-datasets/kedro_datasets/netcdf/netcdf_dataset.py

```diff
@@ -168,11 +169,16 @@ def _save(self, data: xr.Dataset):
                 + "Create an alternate NetCDFDataset with a single .nc output file."
             )
         else:
-            save_path = self._filepath
-            bytes_buffer = data.to_netcdf(**self._save_args)
-
-            with self._fs.open(save_path, mode="wb") as fs_file:
-                fs_file.write(bytes_buffer)
+            if self._protocol == "file":
+                data.to_netcdf(path=self._filepath, **self._save_args)
+            else:
+                assert (
```

Comment:
In Kedro we don't really use `assert` like this, but would instead use exceptions and the `DatasetError` for this.

Reply:
Thanks! Updated this to use `DatasetError`